sdk/storage/storage-blob/src/pollers/BlobStartCopyFromUrlPoller.ts

```diff
@@ -0,0 +1,187 @@
+import { delay } from "@azure/core-http";
+import { PollOperation, PollOperationState, Poller } from "@azure/core-lro";
+import { BlobStartCopyFromURLResponse } from "../generated/src/models";
+import { BlobClient, BlobStartCopyFromURLOptions } from "../BlobClient";
+
+/**
+ * Defines the operations needed from a BlobClient.
+ */
+export type CopyPollerBlobClient = Pick<BlobClient, "abortCopyFromURL" | "getProperties"> & {
+  startCopyFromURL(
+    copySource: string,
+    options?: BlobStartCopyFromURLOptions
+  ): Promise<BlobStartCopyFromURLResponse>;
+};
+
+/**
+ * The state used by the PollOperation.
+ */
+export interface BlobBeginCopyFromUrlPollState
+  extends PollOperationState<BlobStartCopyFromURLResponse> {
+  blobClient: CopyPollerBlobClient;
+  copyId?: string;
+  copyProgress?: string;
+  copySource: string;
+  startCopyFromURLOptions?: BlobStartCopyFromURLOptions;
+}
+
+/**
+ * The PollOperation responsible for:
+ *  - performing the initial startCopyFromURL
+ *  - checking the copy status via getProperties
+ *  - cancellation via abortCopyFromURL
+ */
+export interface BlobBeginCopyFromURLPollOperation
+  extends PollOperation<BlobBeginCopyFromUrlPollState, BlobStartCopyFromURLResponse> {}
+
+/**
+ *
+ */
+export interface BlobBeginCopyFromUrlPollerOptions {
+  blobClient: CopyPollerBlobClient;
+  copySource: string;
+  intervalInMs?: number;
+  onProgress?: (state: BlobBeginCopyFromUrlPollState) => void;
+  resumeFrom?: string;
+  startCopyFromURLOptions?: BlobStartCopyFromURLOptions;
+}
+
+/**
+ *
+ */
+export class BlobBeginCopyFromUrlPoller extends Poller<
+  BlobBeginCopyFromUrlPollState,
+  BlobStartCopyFromURLResponse
+> {
+  public intervalInMs: number;
+
+  constructor(options: BlobBeginCopyFromUrlPollerOptions) {
+    const {
+      blobClient,
+      copySource,
+      intervalInMs = 15000,
+      onProgress,
+      resumeFrom,
+      startCopyFromURLOptions
+    } = options;
+
+    let state: BlobBeginCopyFromUrlPollState | undefined;
+
+    if (resumeFrom) {
+      state = JSON.parse(resumeFrom).state;
+    }
+
+    const operation = makeBlobBeginCopyFromURLPollOperation({
+      ...state,
+      blobClient,
+      copySource,
+      startCopyFromURLOptions
+    });
+
+    super(operation);
+
+    if (typeof onProgress === "function") {
+      this.onProgress(onProgress);
+    }
+
+    this.intervalInMs = intervalInMs;
+  }
+
+  async delay(): Promise<void> {
```

Comment:
>async delay(): Promise<void> { [](start = 2, length = 30)

Why need wrapper the delay method from core-http?

Reply:
We need to implement a delay method in the poller to match the interface. Since we're already importing core-http, makes sense to use the delay helper function from there.